Packs/IBMStorageScale/Integrations/IBMStorageScale/IBMStorageScale.py

```diff
@@ -0,0 +1,546 @@
+import asyncio
+import hashlib
+import time
+from asyncio import Queue
+from typing import Any
+from urllib.parse import urlparse
+
+import httpx
+
+import demistomock as demisto  # noqa: F401
+from CommonServerPython import *  # noqa: F401
+
+# --- CONSTANTS ---
+API_ENDPOINT = "/scalemgmt/v2/cliauditlog"
+PRODUCT = "StorageScale"
+VENDOR = "IBM"
+DEDUPLICATION_WINDOW_MINUTES = 1
+MAX_STORED_HASHES = 10000
+DEFAULT_PAGE_SIZE = 1000  # Default page size for IBM Storage Scale API
+DEFAULT_FIRST_FETCH_MINUTES = 1  # Default days to look back on first fetch
+
+
+# --- UTILITY FUNCTIONS ---
+def generate_event_hash(event: dict[str, Any]) -> str:
+    """
+    Generate a unique hash for an event based on key identifying fields.
+    Uses fields that uniquely identify an event to prevent duplicates.
+    """
+    # Key fields that should uniquely identify an event
+    hash_fields = [
+        event.get("oid", ""),  # Object ID
+        event.get("entryTime", ""),  # Timestamp
+        event.get("user", ""),  # User who executed command
+        event.get("command", ""),  # Command executed
+        event.get("node", ""),  # Node where command was executed
+        event.get("originator", ""),  # Originator of the command
+        event.get("returnCode", ""),  # Return code
+    ]
+
+    # Create a deterministic string from the key fields
+    hash_string = "|".join(str(field) for field in hash_fields)
+
+    # Generate SHA256 hash
+    return hashlib.sha256(hash_string.encode("utf-8")).hexdigest()
+
+
+def get_stored_event_hashes() -> dict[str, str]:
+    """
+    Retrieve stored event hashes from last run object.
+    Returns dict with hash as key and timestamp as value.
+    """
+    last_run = demisto.getLastRun()
+    return last_run.get("event_hashes", {})
+
+
+def store_event_hashes(event_hashes: dict[str, str]) -> None:
+    """
+    Store event hashes in last run object with timestamp cleanup.
+    """
+    # Clean up old hashes outside the deduplication window
+    current_time = datetime.now(timezone.utc)  # Use timezone-aware datetime
+    cutoff_time = (current_time - timedelta(minutes=DEDUPLICATION_WINDOW_MINUTES)).isoformat().replace("+00:00", "Z")
```

Comment:
Perhaps extract replacement of +00:00 and Z (and back) to helper function/s - this could make this more readable